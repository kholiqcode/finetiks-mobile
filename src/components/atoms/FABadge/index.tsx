import React from 'react';

import { Badge } from 'native-base';

import type { IBadgeProps } from 'native-base';

export type FABadgeProps = IBadgeProps;

export default React.memo(function FABadge(props: FABadgeProps) {
  const { testID, ...baseProps } = props;

  return <Badge testID={testID} {...baseProps} />;
});
