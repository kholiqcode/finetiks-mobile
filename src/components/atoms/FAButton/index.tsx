import React from 'react';

import { Button } from 'native-base';

import type { IButtonProps } from 'native-base';

export type FAButtonProps = IButtonProps;

export default React.memo(function FAButton(props: FAButtonProps) {
  const { testID, ...baseProps } = props;

  return <Button testID={testID} {...baseProps} />;
});
