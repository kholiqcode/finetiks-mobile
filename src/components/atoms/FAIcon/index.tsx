import React from 'react';

import { Icon } from 'native-base';
import Feather from 'react-native-vector-icons/Feather';
import FontAwesome from 'react-native-vector-icons/FontAwesome';
import Ionicons from 'react-native-vector-icons/Ionicons';

import type { IIconProps } from 'native-base';

export type FAIconProps = {
  type?: 'font-awesome' | 'ionicons' | 'feather';
} & IIconProps;

export default React.memo(function FAIcon(props: FAIconProps) {
  const { testID, type, ...baseProps } = props;

  let iconType: typeof FontAwesome;

  switch (type) {
    case 'ionicons':
      iconType = Ionicons;
      break;
    case 'feather':
      iconType = Feather;
      break;

    default:
      iconType = FontAwesome;
      break;
  }

  return <Icon testID={testID} as={iconType} {...baseProps} />;
});
