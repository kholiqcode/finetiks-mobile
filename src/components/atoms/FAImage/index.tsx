import React from 'react';

import { Image } from 'native-base';

import type { IImageProps } from 'native-base';

export type FAImageProps = IImageProps;

export default React.memo(function FAImage(props: FAImageProps) {
  const { testID, ...baseProps } = props;

  return <Image testID={testID} {...baseProps} />;
});
