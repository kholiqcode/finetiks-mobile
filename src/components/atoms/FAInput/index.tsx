import React from 'react';

import { Input, InputGroup, InputLeftAddon, InputRightAddon } from 'native-base';

import type { IInputProps } from 'native-base';

export type FAInputProps = {
  leftAddon?: string | JSX.Element;
  rightAddon?: string | JSX.Element;
} & IInputProps;

export default React.memo(function FAInput(props: FAInputProps) {
  const { testID, leftAddon, rightAddon, ...baseProps } = props;

  if (leftAddon || rightAddon) {
    return (
      <InputGroup>
        {leftAddon && <InputLeftAddon children={leftAddon} testID={'leftAddon'} />}
        <Input testID={testID} {...baseProps} placeholder="nativebase" />
        {rightAddon && <InputRightAddon children={rightAddon} testID={'rightAddon'} />}
      </InputGroup>
    );
  }

  return <Input testID={testID} {...baseProps} />;
});
