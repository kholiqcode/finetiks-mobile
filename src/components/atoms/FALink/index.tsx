import React from 'react';

import { Link } from 'native-base';

import type { ILinkProps } from 'native-base';

export type FALinkProps = ILinkProps;

export default React.memo(function FALink(props: FALinkProps) {
  const { testID, ...baseProps } = props;

  return <Link testID={testID} {...baseProps} />;
});
