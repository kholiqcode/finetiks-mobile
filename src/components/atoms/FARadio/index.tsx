import React from 'react';

import { Radio } from 'native-base';

import type { IRadioProps } from 'native-base';

export type IFARadioProps = { label?: string } & IRadioProps;

export default React.memo(function FARadio(props: IFARadioProps) {
  const { testID, label, ...baseProps } = props;

  return (
    <Radio testID={testID} {...baseProps}>
      {label}
    </Radio>
  );
});
