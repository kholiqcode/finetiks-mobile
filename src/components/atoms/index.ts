import FABadge from './FABadge';
import FAButton from './FAButton';
import FAIcon from './FAIcon';
import FAImage from './FAImage';
import FAInput from './FAInput';
import FALink from './FALink';
import FARadio from './FARadio';
import FASelect from './FASelect';
import FASpinner from './FASpinner';
import FASwitch from './FASwitch';
import FATextArea from './FATextArea';

export {
  FABadge,
  FAButton,
  FAIcon,
  FAImage,
  FAInput,
  FALink,
  FARadio,
  FASelect,
  FASpinner,
  FASwitch,
  FATextArea,
};
