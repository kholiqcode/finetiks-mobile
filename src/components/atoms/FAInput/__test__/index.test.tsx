import React from 'react';

import { fireEvent, render } from '@mocks';

import { FAInput } from '@components/atoms';

describe('FAInput', () => {
  it('should render correctly', () => {
    const container = render(<FAInput />);

    expect(container).toBeDefined();
  });

  it('should render with children prop', () => {
    const { getByPlaceholderText } = render(<FAInput size="xs" placeholder="Test Input" />);

    expect(getByPlaceholderText('Test Input')).toBeDefined();
  });

  it('should can change text input', () => {
    const mock = jest.fn();
    const { getByTestId } = render(
      <FAInput testID="test-input" size="xs" placeholder="Test Input" onChangeText={mock} />,
    );

    const inputEmail = getByTestId('test-input');
    expect(inputEmail).toBeDefined();
    fireEvent.changeText(inputEmail, 'email@testing.com');
    expect(mock).toHaveBeenCalledWith('email@testing.com');
  });

  it('should can show leftAddon', () => {
    const leftAddon = 'mock';
    const { getByTestId } = render(<FAInput leftAddon={leftAddon} />);

    expect(getByTestId('leftAddon')).toBeDefined();
  });

  it('should can show rightAddon', () => {
    const rightAddon = 'mock';
    const { getByTestId } = render(<FAInput rightAddon={rightAddon} />);

    expect(getByTestId('rightAddon')).toBeDefined();
  });
});
