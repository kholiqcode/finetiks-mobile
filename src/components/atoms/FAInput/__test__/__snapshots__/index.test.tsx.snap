// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FAInput should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    isFocused={false}
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#d4d4d4",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "overflow": "hidden",
      }
    }
  >
    <TextInput
      accessible={true}
      dataSet={Object {}}
      editable={true}
      fontColor="muted.300"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyPress={[Function]}
      placeholderTextColor="#a3a3a3"
      secureTextEntry={false}
      selectionColor="#1f2937"
      style={
        Object {
          "backgroundColor": "transparent",
          "color": "#d4d4d4",
          "flex": 1,
          "fontFamily": undefined,
          "fontSize": 14,
          "fontStyle": "normal",
          "fontWeight": "500",
          "height": "100%",
          "paddingBottom": 8,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 8,
          "width": "100%",
        }
      }
    />
  </View>
</RNCSafeAreaProvider>
`;
